addons/mail/models/res_partner.py

```diff
@@ -120,7 +120,7 @@ def mail_partner_format(self):
                 "user_id": main_user.id,
                 "is_internal_user": not partner.partner_share,
             }
-            if 'guest' in self.env.context:
+            if self.env.user._is_public():
```

Comment:
So, prior to these changes, a logged user with a guest couldn't acces the email addresses of the partners?